### PR TITLE
Make _init_databricks_cli_config_provider supporting Databricks Runtime 9 ~ 13

### PR DESCRIPTION
**Pull request by WeichenXu123**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/WeichenXu123/mlflow/pull/11198?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11198/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11198
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Make _init_databricks_cli_config_provider supporting Databricks Runtime 9

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [X] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/utils/databricks_utils.py [hunk 1]
```diff
@@ -738,54 +738,107 @@ def _init_databricks_cli_config_provider(entry_point):
...
+                    return DefaultConfigProvider().get_config()
 
-            return DatabricksConfig.from_token(
-                host=api_url, token=api_token, insecure=ssl_trust_all
-            )
+                return DatabricksConfig.from_token(
+                    host=api_url, token=api_token, insecure=ssl_trust_all
+                )
+    elif dbr_major_minor_version >= (9, 0):
+        class DynamicConfigProvider(DatabricksConfigProvider):
```
Comment (WeichenXu123) [inline mlflow/utils/databricks_utils.py:796]: Ref: `spark/driver/python/python_shell/dbruntime/databricksCli.py` in universe 13.1

File: mlflow/utils/databricks_utils.py [hunk 2]
```diff
@@ -738,54 +738,107 @@ def _init_databricks_cli_config_provider(entry_point):
...
+
+                # Fallback to the default behavior if we were unable to find a token.
+                if not api_token_option.isDefined() or not api_url_option.isDefined():
+                    return DefaultConfigProvider().get_config()
+
+                return DatabricksConfig.from_token(
+                    host=api_url_option.get(), token=api_token_option.get(), insecure=ssl_trust_all
+                )
+    else:
+        class DynamicConfigProvider(DatabricksConfigProvider):
```
Comment (WeichenXu123) [inline mlflow/utils/databricks_utils.py:826]: Ref: spark/driver/python/python_shell/dbruntime/databricksCli.py in universe 9.1 <= version <= 10.2

File: mlflow/utils/databricks_utils.py [hunk 3]
```diff
@@ -738,54 +738,107 @@ def _init_databricks_cli_config_provider(entry_point):
...
-
-                ctx = get_context()
-                if ctx and ctx.apiUrl and ctx.apiToken:
-                    return DatabricksConfig.from_token(
-                        host=ctx.apiUrl, token=ctx.apiToken, insecure=ctx.sslTrustAll
+    dbr_version = os.environ["DATABRICKS_RUNTIME_VERSION"]
+    dbr_version_splits = dbr_version.split(".", maxsplit=2)
+    dbr_major_minor_version = (int(dbr_version_splits[0]), int(dbr_version_splits[1]))
+
+    if dbr_major_minor_version >= (13, 2):
```
Comment (BenWilson2) [inline mlflow/utils/databricks_utils.py:745]: Do we need a guard here if the dbr_major_minor_version doesn't resolve to a tuple of int?

Comment (WeichenXu123) [reply mlflow/utils/databricks_utils.py:745]: what's the case of not a tuple of int ? 
and how to handle it ?

Comment (BenWilson2) [reply mlflow/utils/databricks_utils.py:745]: Right now we'll get a really confusing error message if dbr_version doesn't fetch. Should we raise if it doesn't return a splittable string?

Comment (WeichenXu123) [reply mlflow/utils/databricks_utils.py:745]: OK. sounds good!

**Review by BenWilson2 — APPROVED** [review]
LGTM! Thanks Weichen :)

**Pull request merged**